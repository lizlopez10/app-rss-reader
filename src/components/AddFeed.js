import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddFeed = ({ setFeeds }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setFeeds(feeds => [inputValue, ...feeds,]);
            setInputValue('');
        }

    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


AddFeed.propTypes = {
    setFeeds: PropTypes.func.isRequired
}
