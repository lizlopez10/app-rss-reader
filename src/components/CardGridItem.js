import React from 'react';
import imgNotFound from '../assets/img/NoImage.png';
export const CardGridItem = ( {title, main_link, main_image, items=[] } ) => {    
    
    const list_links = ({link, title})=>{
        
        return (
            <a className="card-text" href={link} target="_blank" title={title}>
                <p className="card-text" style={{ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' }}>{title}</p>
            </a>
        );
    }
    return (
        <div className="card" style={{ width: '18rem' }}>
            <img src={main_image || imgNotFound} alt={title} className="card-img-top" />
            <div className="card-body">
                <h5 className="card-title">{title}</h5>
                {items.slice(0, 5).map(item => list_links(item))}
                <a href={main_link} className="btn btn-primary text-center" target="_blank">Ver mÃ¡s</a>
            </div>
        </div>
    )
}
