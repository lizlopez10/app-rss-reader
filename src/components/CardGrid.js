import React from 'react';
import { useFetchFeed} from '../hooks/useFetchFeed';
import { CardGridItem } from './CardGridItem';

export const CardGrid = ({ feed_url }) => {
    
    try {        
    
        const { data:feed, loading } = useFetchFeed( feed_url );        
        return (
            <>            
                { loading && <p className="animate__animated animate__flash">Loading</p> }                            
                    {                    
                            <CardGridItem 
                                key={ feed.title }
                                {...feed }
                            />                    
                    }            
                
            </>
        )
        
    } catch (error) {
            throw error;
    }
}
