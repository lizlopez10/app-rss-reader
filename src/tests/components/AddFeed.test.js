import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { AddFeed } from '../../components/AddFeed';

// const mockSetFeed = 'http://feeds.washingtonpost.com/rss/rss_compost?itid=lk_inline_manual_10';
const mockSetFeed = jest.fn();
const handleInputChange = jest.fn();

describe('Test in <AddFeed />', () => {

  const wrapper = mount(<AddFeed setFeeds={mockSetFeed} />);

  test('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test('should trigger action handleInputChange', () => {
  //   const e = { target: { value: 'test' } };
  //   act( () => {
  //     wrapper2.find('input').prop('onChange')(e);
  //     expect(handleInputChange).toHaveBeenCalled();
  //   });
  // });

});