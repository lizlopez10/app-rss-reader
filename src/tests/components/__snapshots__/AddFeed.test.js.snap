// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddFeed /> should show component correctly 1`] = `
<AddFeed
  setFeeds={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</AddFeed>
`;
